### ISSUE TITLE
Node cannot be cast to java.util.HashMap$TreeNode

### ISSUE THREAD

**Issue by cdsap** [cdsap/Talaiot#192]
We are using version `1.3.1` and we have seen in some builds(less than 1%):
```
java.lang.ClassCastException: java.util.HashMap$Node cannot be cast to java.util.HashMap$TreeNode
	at com.cdsap.talaiot.BuildCacheOperationListener.finished(BuildCacheOperationListener.kt:59)
	at org.gradle.internal.operations.DefaultBuildOperationListenerManager$ProgressShieldingBuildOperationListener.finished(DefaultBuildOperationListenerManager.java:128)
	...  
```

The step has high pressure of memory and maybe it's not related with the library the error.

**Comment by MyDogTom** [comment]
Looks like it was a bad idea to keep `ExecuteTaskBuildOperationType.Result` since we don't have control on this structure and  the size of it. That's might be the reason for the high memory pressure. I suggest to create our own structure tailored to our needs. My expectation is that such structure will have smaller memory footprint. I will open a PR with a proposal.

@cdsap Do you know which tool I could use to measure the difference? Would be nice prove that there is a real impact. Perhaps https://github.com/gradle/gradle-profiler/ might help with this, but I've never used this tool.

**Comment by cdsap** [comment]
in terms of the error it's fair to say it's happening not very often. We are having around 200 daily executions and I've seen in no more than 10 steps overall in 2 weeks. 

I really recommend Gradle Profiler. I have sometime problems when it's executed in containers but it's quite flexible. 
For regressions in changes of the build I use https://github.com/cdsap/Bagan It requires cloud k8s infrastructure. Let me know if you want to know more about Bagan.

**Comment by cdsap** [comment]
fixed with #194
